File: Python/foobar_bench.py
#! /usr/bin/env python
#
# a little, very inaccurate benchmark tool
import time

__author__ = "Landei"
__email__ = "website@landei.pfweb.eu"
__version__ = "1.0"
__date__ = "28.11.2014"

eingabe = input("Input number range: ")
#int zahl, ziffer, ziffer2, eingabe
zahl = 0
t1 = time.time()
ziffer = zahl % 5
ziffer2 = zahl % 3

while zahl < eingabe:
    zahl = zahl + 1
    ziffer = zahl % 5
    ziffer2 = zahl % 3
    
    if (ziffer !=0) and (ziffer2 !=0):
        print(zahl)
    elif (ziffer == 0) and (ziffer2 == 0):
        print("foobar")
    elif (ziffer == 0):
        print("bar")
    elif (ziffer2 == 0):
        print("foo")

t2 = time.time()
timeEnd = (t2 - t1)

print "Time required: "+str(timeEnd)+" sek.\n"

File: Java/foobar_bench.java
/**
  *
  * Beschreibung
  *
  * @version 1.0 (28.11.2014)
  * @Landei 
  */

public class foobar_bench {
  
  public static void main(String[] args) {
    String input, error;  
    do {
      
      int zahl, ziffer, ziffer2, eingabe;
      double timeStart, timeEnd;
      
      zahl = 0;
      timeStart = 0;
      timeEnd = 0;
      
      eingabe = InOut.readInt("Input number range: ");
      
      do {
        timeStart = System.currentTimeMillis();
        
        ++zahl;
        ziffer = zahl % 5;
        ziffer2 = zahl % 3;
        
        if (ziffer !=0 && ziffer2 !=0) {
          System.out.println(zahl);
        } // end of if
        
        else if(ziffer == 0 && ziffer2 == 0){
          System.out.println("foobar");
        } // end of if-else
        
        else if(ziffer == 0){
          System.out.println("bar");
        } // end of if-else
        
        else if(ziffer2 == 0){
          System.out.println("foo");
        } // end of if-else
        
      } while (zahl < eingabe); // end of do-while
      
      timeEnd = System.currentTimeMillis();
      
      System.out.println("Time required: " +((timeEnd - timeStart)/1000) +"sek.");
      
      input = InOut.readString("Try again? (y/n)");
      
      if (input.equals("n")) System.exit(0);
    } while (input.equals("y")); // end of do-while  
  } // end of main
  
} // end of class foobar-bench
